hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/analysis/HoodieAnalysis.scala

```diff
@@ -46,6 +46,11 @@ object HoodieAnalysis extends SparkAdapterSupport {
   def customResolutionRules: Seq[RuleBuilder] = {
     val rules: ListBuffer[RuleBuilder] = ListBuffer()
 
+    if (HoodieSparkUtils.gteqSpark3_5) {
```

Comment:
this rule should be applied after the `HoodieSpark35DataSourceV2ToV1Fallback` rule. Before applying RelationInsertInto again, the v2Relation has already been fallback to v1Relation.

Reply:
Makes sense.  Fixed.